tests/main/snap-run-hook/task.yaml

```diff
@@ -0,0 +1,69 @@
+summary: Check that `snap run` can actually run hooks
+
+prepare: |
+    echo "Ensure we have a core snap with snap run"
+    $SPREAD_PATH/tests/lib/reset.sh
+    snap install --channel=candidate ubuntu-core
+    snapbuild $TESTSLIB/snaps/basic-hooks .
+    snap install basic-hooks_1.0_all.snap
+
+    # Use the in-tree snap-exec, not the one in the core snap. We'll do that via
+    # overlayfs and a copy.
+    tmp=$(mktemp -d)
+    overlay="$tmp/overlay"
+    work="$tmp/work"
+    mkdir "$overlay" "$work"
+    to_overlay="$(realpath /snap/ubuntu-core/current)"
+    mount -t overlayfs -o lowerdir=$to_overlay,upperdir=$overlay,workdir=$work overlayfs $to_overlay
+    cp /usr/lib/snapd/snap-exec $to_overlay/usr/lib/snapd/
```

Comment:
I wonder if we should have something in the test to double check that the snap-exec we use is really the one we copied here. Maybe as simple as just doing a `cmp /usr/lib/snapd/snap-exec /snap/ubuntu/-core/current/usr/lib/snapd/snap-exec || (echo "snap-exec overlay did not work"; exit 1)` ?